### ISSUE TITLE
Wrong error message at GET /v2/entities/Room1 when 'attrs=' uses unkown attribute

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#2241]
Not sure if already covered in another existing issue  (@iariasleon could you have a look?)...

The following operation (on an existing entity) is reporting wrong error message, i.e. the attribute is the element that doesn't exist, not the entity:

```
curl localhost:1026/v2/entities/Room1?attrs=mistaken

{"error":"NotFound","description":"The requested entity has not been found. Check type and id"}
```

Note that a similar operation shows the right message:

```
curl localhost:1026/v2/entities/Room1/attrs/mistake

{"error":"NotFound","description":"The entity does not have such an attribute"}
```

Once this gets fixed in Orion, NGSIv2 cookbook should be updated to re-include the "One entity attribute. Not present" operation (cc: @FiwareULPGC )

**Comment by iariasleon** [comment]
Yes, see https://github.com/telefonicaid/fiware-orion/issues/1769 issue

**Comment by kzangeli** [comment]
During functests, we've seen a strange difference in mongo queries between two very similar requests:

1. PUT /v2/entities/E1?attrs=A2
2. PUT /v2/entities/E1/attrs/A2

The mongo queries for these two requests are:

    1. { query: { $or: [ { _id.id: "E1" } ], _id.servicePath: { $in: [ /^/.*/, null ] }, attrNames: { $in: [ "A2" ] } }, orderby: { creDate: 1 } }
    2. { query: { $or: [ { _id.id: "E1" } ], _id.servicePath: { $in: [ /^/.*/, null ] } }, orderby: { creDate: 1 } }

Query No 1 includes A2 in the query, so when nothing is found, we don't know whether E1 exists.
Query No 2 is not very efficient as it will extract ALL attributes of E1 from mongo (there is no limit on how many attributes an entity can have)  when it is really only interested in A2. However, thanks to this inefficient query, we always know if the entity exists or not.

Problem number one here is, why does the broker behave in two different ways for two queries that are so extremely similar?
The answer may lie in the service routines and how they build their respective QueryContextRequest that is passed to mongoQueryContext().

**Comment by kzangeli** [comment]
A proposal (provoking ~20 errors in functests) on 'only' new string describing that the combination of entity id and attribute name was not found in the branch `bug/2241_error_response_on_attr_not_found`, in origin.

This branch `bug/2241_error_response_on_attr_not_found` **should be removed** after this issue is closed.

**Comment by fgalan** [comment]
To follow RESTful principles, 404 Not Found error should be used only in the case of resource not found. Note that resource is identified by URL path, but not URL params, such as `?attrs=`.

Thus, in the case of attribute not existing in existing entity with the operation

`GET /v2/entities/E1/attrs/A2`

it is clear that the result should be 404 Not Found (as the current behaviour is). However, in the same case attribute not existing in existing entity the following operations:

```
GET /v2/entities/E1?attrs=A2
GET /v2/entities/E1/attrs?attrs=A2
```

404 Not Found **must not** be used, as the resource associated to the operation (i.e., the URL path, either `/v2/entities/E1` or `/v2/entities/E1/attrs`, does exist).

Thus, what do do when after applying the `?attrs=` filter the result is "empty set" for attributes. In order to follow response syntax it should be as follows:

```
GET /v2/entities/E1?attrs=A2

200 OK
{ "id": "E1", "type": ... }
```

and

```
GET /v2/entities/E1/attrs?attrs=A2

200 OK
{}
```

**Comment by fgalan** [comment]
Fixed in PR #2321
